### PR TITLE
tests: RabbitMQ as a default BROKER_URL

### LINKED ISSUE #40: tests: conflict between CliRunner and Celery in memory broker
When I run  the tests I get the exception:
```
tests/test_utils.py:105:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
invenio_indexer/api.py:195: in process_bulk_queue
    request_timeout=req_timeout,
../../lib/python3.5/site-packages/elasticsearch/helpers/__init__.py:190: in bulk
    for ok, item in streaming_bulk(client, actions, **kwargs):
../../lib/python3.5/site-packages/elasticsearch/helpers/__init__.py:162: in streaming_bulk
    for result in _process_bulk_chunk(client, bulk_actions, raise_on_exception, raise_on_error, **kwargs):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

client = <LocalProxy unbound>, bulk_actions = ['{"delete": {"_index": "records-default-v1.0.0", "_type": "default-v1.0.0", "_id": "7133d6e5-a332-455d-b0c8-973ca420f...ete": {"_index": "records-default-v1.0.0", "_type": "default-v1.0.0", "_id": "7133d6e5-a332-455d-b0c8-973ca420f686"}}'], raise_on_exception = True, raise_on_error = True
kwargs = {'request_timeout': 10}, errors = [{'delete': {'_id': '7133d6e5-a332-455d-b0c8-973ca420f686', '_index': 'records-default-v1.0.0', '_shards': {'failed': 0, 'successful': 1, 'total': 2}, '_type': 'default-v1.0.0', ...}}], resp = {'errors': False, 'items': [{}, {}], 'took': 7}

    def _process_bulk_chunk(client, bulk_actions, raise_on_exception=True, raise_on_error=True, **kwargs):

[...TRUNCATED FOR BREVITY...]

        if errors:
>           raise BulkIndexError('%i document(s) failed to index.' % len(errors), errors)
E           elasticsearch.helpers.BulkIndexError: ('1 document(s) failed to index.', [{'delete': {'found': False, '_version': 2, '_shards': {'successful': 1, 'total': 2, 'failed': 0}, '_id': '7133d6e5-a332-455d-b0c8-973ca420f686', 'status': 404, '_type': 'default-v1.0.0', '_index': 'records-default-v1.0.0'}}])

../../lib/python3.5/site-packages/elasticsearch/helpers/__init__.py:134: BulkIndexError
```

After some more digging it looks like two delete messages are queued [when one is published here](https://github.com/inveniosoftware/invenio-indexer/blob/0dce9c471dde7830cbcab15ee2b6356039f80e0e/invenio_indexer/utils.py#L54)

This only happens when `tests/test_cli.py::test_reindex` is ran before `tests/test_utils.py::test_record_indexing` like this:

` py.test tests/test_cli.py::test_reindex tests/test_utils.py::test_record_indexing`

If `tests/test_utils.py::test_record_indexing` alone is ran then the issue does not appear.

I tracked down the issue to the `BROCKER_URL` config variable. The problem arises only with `BROKER_URL=os.environ.get('BROKER_URL', 'memory://')` but not with rabbitmq or redis.

### PR DESCRIPTION
**Pull request by nharraud**
* FIX Fixes tests by using rabbitmq instead of in memory queue.
  (closes #40)

Signed-off-by: Nicolas Harraudeau <nicolas.harraudeau@cern.ch>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def app(request):
     instance_path = tempfile.mkdtemp()
     app = Flask('testapp', instance_path=instance_path)
     app.config.update(
-        BROKER_URL=os.environ.get('BROKER_URL', 'memory://'),
```
Comment (jirikuncar) [inline tests/conftest.py]: Can you just change the default value so one can still test with different brokers?

**Comment by jirikuncar** [comment]
@nharraud WDYT about new PR title as commit title?

**Comment by nharraud** [comment]
I would just specify in the title that it is only for the tests.

**Comment by jirikuncar** [comment]
it's clear from the component name ;)

**Comment by nharraud** [comment]
@jirikuncar you're right. Fixing it.

**Pull request closed without merging**